README.md

```diff
@@ -1,2 +1,31 @@
-# bubo
-memory efficient node.js object cache
+# Object Hash Set
+This is a hash set for Javascript objects. Its object-encoding algorithm is quite space-efficient, especially in the case where the input objects have largely the same keys, and most or all keys have low cardinality (number of distinct values of that key in the data set). Data with these characteristics often arises in time series data analysis applications, which is where this originated.
+
+## API
+
+### new ObjectHashSet([options]) ###
+Creates an instance of the Object Hash Set. `options`, if specified, is an object. The only supported option is `ignore`. The value of `ignore` is an array of keys that the set will not pay attention to during storage or lookup. The set will consider two objects identical if their values for all non-ignored keys are the same.
+
+### add(object) ###
+If `add` has already been called on this Object Hash Set with an object whose non-ignored key-value pairs are the same as the given object, returns `false`. Otherwise, returns `true` and stores the object so that future `add`s will return `false`.
+
+### contains(object) ###
+Returns `true` if an object equivalent to `object` has already been `add`ed.
+### delete(object) ###
+The next call to `add` or `contains` with an object identical to the given object will return `false`.
```

Comment:
Again this seems oddly phrased and should mention about how it doesn't actually free up the memory. How about:

"Removes the object from the set. A future call to `add` or `contains` with an object identical to the given object will return `false`. Note that this will not reclaim the storage space used by the keys in the given object."